### PR TITLE
Add better warning for misspelled files

### PR DESCRIPTION
**Pull request by morganchen12**

### CODE REVIEW

File: FirebaseCore/Sources/FIRApp.m [hunk 1]
```diff
@@ -883,4 +886,33 @@ - (void)appDidBecomeActive:(NSNotification *)notification {
   }
 }
 
+#if DEBUG
++ (void)findMisnamedGoogleServiceInfoPlist {
+  for (NSBundle *bundle in [NSBundle allBundles]) {
+    // Not recursive, but we're looking for misnames, not people accidentally
+    // hiding their config file in a subdirectory of their bundle.
+    // This is only really designed for iOS.
```
Comment (andrewheard) [inline FirebaseCore/Sources/FIRApp.m:894]: Any reason you suspect this won't work on other platforms? I _think_ `pathsForResourcesOfType:inDirectory:` searches the `Resources` subdirectory on macOS (where the `GoogleService-Info.plist` file lives) when `inDirectory` is `nil` based on the [discussion](https://developer.apple.com/documentation/foundation/nsbundle/1413058-pathsforresourcesoftype#discussion) in the docs.

**Review by andrewheard — APPROVED** [review]
Love this, thanks!

File: FirebaseCore/Sources/FIRApp.m (hunk 1 above)
Comment (morganchen12) [reply FirebaseCore/Sources/FIRApp.m:894]: It should work for all platforms, but macOS bundles may make use of subdirectories more frequently. I can remove this line if it's misleading.

**Comment by paulb777** [comment]
Please fix ci

**Comment by morganchen12** [comment]
Tested locally and it works as expected.

```
2023-05-23 13:43:13.340213-0700 AuthenticationExample[49110:6833337] *** Terminating app due to uncaught exception 'com.firebase.core', reason: '`FirebaseApp.configure()` could not find the default configuration plist in your project, but did find one at /Users/morganchen/Library/Developer/CoreSimulator/Devices/21552B4A-FC19-456F-AB44-D7A588594994/data/Containers/Bundle/Application/CD377DAB-0E09-4D4B-98F4-1C79EBA6245E/AuthenticationExample.app/GoogleService-Info .plist. Please rename this file to GoogleService-Info.plist to use it as the default configuration.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007ff800427378 __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007ff80004dbaf objc_exception_throw + 48
	2   CoreFoundation                      0x00007ff800427256 -[NSException initWithCoder:] + 0
	3   FirebaseCore                        0x0000000108cdda6d +[FIRApp configure] + 813
	4   AuthenticationExample               0x000000010870d6d9 -[AppDelegate application:didFinishLaunchingWithOptions:] + 89
	5   UIKitCore                           0x000000010edeb193 -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 271
	6   UIKitCore                           0x000000010edecf34 -[UIApplication _callInitializationDelegatesWithActions:forCanvas:payload:fromOriginatingProcess:] + 4257
	7   UIKitCore                           0x000000010edf2bf5 -[UIApplication _runWithMainScene:transitionContext:completion:] + 1236
	8   UIKitCore                           0x000000010e22f19c -[_UISceneLifecycleMultiplexer completeApplicationLaunchWithFBSScene:transitionContext:] + 122
	9   UIKitCore                           0x000000010e89e9f5 _UIScenePerformActionsWithLifecycleActionMask + 88
	10  UIKitCore                           0x000000010e22fc96 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke + 198
	11  UIKitCore                           0x000000010e22f6d3 -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:] + 252
	12  UIKitCore                           0x000000010e22facc -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 831
	13  UIKitCore                           0x000000010e22f379 -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:] + 354
	14  UIKitCore                           0x000000010e23e1a0 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke + 178
	15  UIKitCore                           0x000000010e76fe53 +[BSAnimationSettings(UIKit) tryAnimatingWithSettings:fromCurrentState:actions:completion:] + 833
	16  UIKitCore                           0x000000010e8bca90 _UISceneSettingsDiffActionPerformChangesWithTransitionContextAndCompletion + 257
	17  UIKitCore                           0x000000010e23de20 -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:] + 348
	18  UIKitCore                           0x000000010e022523 __64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke.595 + 815
	19  UIKitCore                           0x000000010e020f39 -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:] + 245
	20  UIKitCore                           0x000000010e0220d0 -[UIScene scene:didUpdateWithDiff:transitionContext:completion:] + 255
	21  UIKitCore                           0x000000010edf143b -[UIApplication workspace:didCreateScene:withTransitionContext:completion:] + 733
	22  UIKitCore                           0x000000010e7a27e0 -[UIApplicationSceneClientAgent scene:didInitializeWithEvent:completion:] + 350
	23  FrontBoardServices                  0x00007ff805445c8a -[FBSScene _callOutQueue_agent_didCreateWithTransitionContext:completion:] + 415
	24  FrontBoardServices                  0x00007ff805473deb __92-[FBSWorkspaceScenesClient createSceneWithIdentity:parameters:transitionContext:completion:]_block_invoke.187 + 102
	25  FrontBoardServices                  0x00007ff805453cda -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 209
	26  FrontBoardServices                  0x00007ff8054739e0 __92-[FBSWorkspaceScenesClient createSceneWithIdentity:parameters:transitionContext:completion:]_block_invoke + 344
	27  libdispatch.dylib                   0x00000001096bef5b _dispatch_client_callout + 8
	28  libdispatch.dylib                   0x00000001096c28d2 _dispatch_block_invoke_direct + 496
	29  FrontBoardServices                  0x00007ff80549a6c0 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 30
	30  FrontBoardServices                  0x00007ff80549a5b6 -[FBSSerialQueue _targetQueue_performNextIfPossible] + 174
	31  FrontBoardServices                  0x00007ff80549a6e8 -[FBSSerialQueue _performNextFromRunLoopSource] + 19
	32  CoreFoundation                      0x00007ff800386ebd __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	33  CoreFoundation                      0x00007ff800386dfc __CFRunLoopDoSource0 + 157
	34  CoreFoundation                      0x00007ff8003865f9 __CFRunLoopDoSources0 + 212
	35  CoreFoundation                      0x00007ff800380db3 __CFRunLoopRun + 927
	36  CoreFoundation                      0x00007ff800380637 CFRunLoopRunSpecific + 560
	37  GraphicsServices                    0x00007ff809c0f28a GSEventRunModal + 139
	38  UIKitCore                           0x000000010edef425 -[UIApplication _run] + 994
	39  UIKitCore                           0x000000010edf4301 UIApplicationMain + 123
	40  AuthenticationExample               0x000000010870de28 main + 104
	41  dyld                                0x0000000108a0d2bf start_sim + 10
	42  ???                                 0x000000010dec541f 0x0 + 4528559135
)
libc++abi: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'com.firebase.core', reason: '`FirebaseApp.configure()` could not find the default configuration plist in your project, but did find one at /Users/morganchen/Library/Developer/CoreSimulator/Devices/21552B4A-FC19-456F-AB44-D7A588594994/data/Containers/Bundle/Application/CD377DAB-0E09-4D4B-98F4-1C79EBA6245E/AuthenticationExample.app/GoogleService-Info .plist. Please rename this file to GoogleService-Info.plist to use it as the default configuration.'
terminating with uncaught exception of type NSException
CoreSimulator 857.7 - Device: iPhone 14 Pro (21552B4A-FC19-456F-AB44-D7A588594994) - Runtime: iOS 16.0 (20A360) - DeviceType: iPhone 14 Pro
```

**Review by paulb777 — APPROVED** [review]

**Pull request merged**